src/sentry/api/endpoints/organization_searches.py

```diff
@@ -75,3 +86,28 @@ def get(self, request, organization):
             ).order_by('name', 'project'))
 
         return Response(serialize(results, request.user))
+
+    def post(self, request, organization):
+        serializer = OrganizationSearchSerializer(data=request.DATA)
+
+        if serializer.is_valid():
+            result = serializer.object
+            # Prevent from creating duplicate queries
+            if SavedSearch.objects.filter(
+                Q(is_global=True) | Q(organization=organization, owner__isnull=True),
+                query=result['query'],
+            ).exists():
+                return Response(
+                    {'detail': 'Query {} already exists'.format(result['query'])},
```

Comment:
```suggestion
                    {'detail': u'Query {} already exists'.format(result['query'])},
```

Potential UnicodeError :ghost: